application/src/main/java/org/opentripplanner/routing/algorithm/mapping/StatesToWalkStepsMapper.java

```diff
@@ -515,6 +522,33 @@ private WalkStepBuilder createElevatorWalkStep(State backState, State forwardSta
     return step;
   }
 
+  private WalkStepBuilder createStationEntranceWalkStep(
+    State backState,
+    State forwardState,
+    Edge edge
+  ) {
+    // don't care what came before or comes after
+    var step = createWalkStep(forwardState, backState);
+
+    step.withRelativeDirection(RelativeDirection.ENTER_OR_EXIT_STATION);
```

Comment:
You could add a comment here to explain that figuring out if something is an entrance or exit is difficult here.